tests/foreman/api/test_syncplan.py

```diff
@@ -397,11 +397,12 @@ def test_positive_update_interval_custom_cron(self):
         :CaseImportance: Critical
         """
         for interval in valid_sync_interval():
-            sync_plan = entities.SyncPlan(
-                description=gen_string('alpha'),
-                organization=self.org,
-                interval=interval
-            ).create()
+            if interval != 'custom cron':
+                sync_plan = entities.SyncPlan(
+                    description=gen_string('alpha'),
+                    organization=self.org,
+                    interval=interval
+                ).create()
```

Comment:
The other code should be in the if interval != 'custom cron': block

```python
        for interval in valid_sync_interval():
            if interval != 'custom cron':
                sync_plan = entities.SyncPlan(
                    description=gen_string('alpha'),
                    organization=self.org,
                    interval=interval
                ).create()
    
                sync_plan.interval = SYNC_INTERVAL['custom']
                sync_plan.cron_expression = gen_choice(valid_cron_expressions())
                self.assertEqual(
                    sync_plan.update(['interval', 'cron_expression']).interval,
                    SYNC_INTERVAL['custom']
                )
```

Reply:
or (best option) we rebuild the valid_sync_intervals
```python
        valid_intervals = [interval for interval in valid_sync_interval()
                           if interval != SYNC_INTERVAL['custom']]
        for interval in valid_intervals:
            sync_plan = entities.SyncPlan(
                description=gen_string('alpha'),
                organization=self.org,
                interval=interval
            ).create()

            sync_plan.interval = SYNC_INTERVAL['custom']
            sync_plan.cron_expression = gen_choice(valid_cron_expressions())
            self.assertEqual(
                sync_plan.update(['interval', 'cron_expression']).interval,
                SYNC_INTERVAL['custom']
            )
```

Reply:
@ldjebran I prefer the first option, since there really isn't a need to iterate over the valid_sync_intervals list twice. Also, it is much more readable.